### ISSUE TITLE
use PB to predict on ios

### ISSUE THREAD

**Issue by 1372072273** [lighttransport/prnet-infer#3]
I use the transform code to pb,and the use it in ios,but it get the question: libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Seesion run failure: Attempting to use uninitialized value resfcn256/Conv/BatchNorm/moving_mean
	 [[Node: resfcn256/Conv/BatchNorm/FusedBatchNorm = FusedBatchNorm[T=DT_FLOAT, data_format="NHWC", epsilon=0.001, is_training=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](resfcn256/Conv/Conv2D, Const_1, Const_2, resfcn256/Conv/BatchNorm/moving_mean, resfcn256/Conv/BatchNorm/moving_variance)]]...  I use https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/graph_transforms/README.md#fold_old_batch_norms to transform the pb and remove the BatchNorm nodes,but  it doesn't work,can you help me?

**Comment by syoyo** [comment]
I don't know libtensorflow works well on iOS.

You are better to convert .pb model to tflite.

We've confirmed tflite version of PRNet pretty works well on iOS and Android(25 ms on iPhone11 Pro for single image inference)

**Comment by 1372072273** [comment]
> I don't know libtensorflow works well on iOS.
> 
> You are better to convert .pb model to tflite.
> 
> We've confirmed tflite version of PRNet pretty works well on iOS and Android(25 ms on iPhone11 Pro for single image inference)

what's the version of tensorflow?I tried the tensorflow 2.0,1.9,1.8,but the question is tf 1.x don't have the  function TFLiteConverter ,tf 2.0 had the question  that can't  find the saved model,i found that tf 2.0 can only use saved model and freezen graph,so i want to know what's your version of tf?

**Comment by 1372072273** [comment]
what's the version of tensorflow?I tried the tensorflow 2.0,1.9,1.8,but the question is tf 1.x don't have the function TFLiteConverter ,tf 2.0 had the question that can't find the saved model,i found that tf 2.0 can only use saved model and freezen graph,so i want to know what's your version of tf?

**Comment by 1372072273** [comment]
> I don't know libtensorflow works well on iOS.
> 
> You are better to convert .pb model to tflite.
> 
> We've confirmed tflite version of PRNet pretty works well on iOS and Android(25 ms on iPhone11 Pro for single image inference)

what's the version of tensorflow?I tried the tensorflow 2.0,1.9,1.8,but the question is tf 1.x don't have the function TFLiteConverter ,tf 2.0 had the question that can't find the saved model,i found that tf 2.0 can only use saved model and freezen graph,so i want to know what's your version of tf?

**Comment by syoyo** [comment]
We have used tf `r1.14` and `r1.15`

**Comment by syoyo** [comment]
`r1.12` would work, but not recommended.

https://github.com/lighttransport/prnet-infer#dependences

**Comment by 1372072273** [comment]
> `r1.12` would work, but not recommended.
> 
> https://github.com/lighttransport/prnet-infer#dependences

thanks for your help

**Comment by 1372072273** [comment]
> `r1.12` would work, but not recommended.
> 
> https://github.com/lighttransport/prnet-infer#dependences

I had found what's question in my pb,I forgot to freeze the PB when I use the  code to write graph in function predict,after freeze the graph i can use the pb to predict on python,thank you!

**Comment by syoyo** [comment]
Yes, you need to create freezed model for the inference.

so is it ok to close this issue? > @1372072273

**Comment by 1372072273** [comment]
yes,thank you
